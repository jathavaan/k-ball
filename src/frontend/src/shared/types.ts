export interface Player {
  playerId: number;
  fullName: string;
  currentClub: string;
  imageUrl: string;
}
