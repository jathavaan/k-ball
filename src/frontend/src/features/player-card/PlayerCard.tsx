import { CardText, ImageContainer, Text } from "@features/ui";
import Grid from "@mui/material/Grid2";
import { CardActionArea } from "@mui/material";
import { Link as RouterLink } from "react-router-dom";
import { StyledPlayerCard } from "@features/player-card/playerCard.style.ts";
import { PlayerCardProps } from "@features/player-card/playerCard.types.ts";
import PublicIcon from "@mui/icons-material/Public";
import CakeIcon from "@mui/icons-material/Cake";
import FmdGoodIcon from "@mui/icons-material/FmdGood";
import StarIcon from "@mui/icons-material/Star";

export const PlayerCard = (props: PlayerCardProps) => {
  return (
    <StyledPlayerCard data-testid="player-card">
      <CardActionArea component={RouterLink} to={`${props.playerId}`}>
        <Grid
          container
          sx={{
            position: "relative",
            minHeight: "8rem",
            overflow: "hidden",
          }}
          spacing={0}
        >
          <Grid
            size={{ xs: 6 }}
            sx={() => ({
              position: "relative",
              clipPath: "polygon(0 0, 90% 0, 100% 100%, 0 100%)",
              backgroundColor: "#c2c2c2",
              zIndex: 1,
            })}
          >
            <ImageContainer
              alt={`Profile picture for ${props.fullName}`}
              src={props.imageUrl}
              style={{
                width: "100%",
                height: "100%",
                objectFit: "cover",
                borderRadius: "0",
              }}
            />
          </Grid>
          <Grid
            container
            spacing={1}
            size={{ xs: 6 }}
            sx={{
              display: "flex",
              alignItems: "flex-start",
              flexDirection: "column",
              justifyContent: "space-between",
              marginBottom: "0.1rem",
            }}
          >
            <Grid
              size={{ xs: 12 }}
              sx={(theme) => ({
                display: "flex",
                flexDirection: "column",
                alignItems: "left",
                justifyContent: "space-between",
                backgroundColor: theme.palette.secondary.dark,
                borderBottomRightRadius: "0.4rem",
                borderBottomLeftRadius: 0,
                borderTopLeftRadius: 0,
                position: "relative",
                left: "-10%",
                width: "110%",
                minHeight: {
                  xs: "1.8rem",
                  md: "2.5rem",
                },
              })}
            >
              <Text
                text={props.fullName}
                sx={(theme) => ({
                  color: theme.palette.secondary.contrastText,
                  fontSize: { xs: "1rem", md: "1.1rem" },
                  fontWeight: 300,
                  textAlign: "left",
                  marginLeft: "0.6rem",
                })}
              />
              <Text
                text={
                  props.averageRating
                    ? `Rating: ${parseFloat(props.averageRating.toString()).toFixed(0)}/5`
                    : `Be the first to rate!`
                }
                sx={(theme) => ({
                  fontSize: {
                    xs: "0.7rem",
                    md: "0.8rem",
                  },
                  color: theme.palette.grey["300"],
                  textAlign: "left",
                  marginLeft: "0.6rem",
                })}
              />
            </Grid>
            <Grid
              size={{ xs: 12 }}
              sx={{
                display: "flex",
                flexDirection: "row",
                position: "relative",
                alignItems: "center",
                left: "-8%",
              }}
            >
              <CardText
                icon={
                  <ImageContainer
                    src={props.clubLogoUrl}
                    alt={`${props.currentClub} logo`}
                    style={{
                      height: "1.2rem",
                      width: "1.2rem",
                    }}
                  />
                }
                text={props.currentClub}
              />
            </Grid>
            <Grid
              size={{ xs: 12 }}
              sx={{
                display: "flex",
                flexDirection: "row",
                position: "relative",
                alignItems: "center",
                left: "-6%",
              }}
            >
              <CardText
                icon={
                  <FmdGoodIcon
                    sx={{
                      fontSize: { xs: "0.9rem", md: "1rem" },
                    }}
                  />
                }
                text={props.position}
              />
            </Grid>
            <Grid
              size={{ xs: 12 }}
              sx={{
                display: "flex",
                flexDirection: "row",
                position: "relative",
                alignItems: "center",
                left: "-4%",
              }}
            >
              <CardText
                icon={
                  <CakeIcon
                    sx={{
                      fontSize: { xs: "0.9rem", md: "1rem" },
                    }}
                  />
                }
                text={`${Intl.DateTimeFormat("en-US", {
                  //month: "short",
                  //day: "numeric",
                  year: "numeric",
                }).format(new Date(props.birthDate))} | ${props.age} years`}
              />
            </Grid>
            <Grid
              size={{ xs: 12 }}
              sx={{
                display: "flex",
                flexDirection: "row",
                position: "relative",
                alignItems: "center",
                left: "-2%",
              }}
            >
              <CardText
                icon={
                  <PublicIcon
                    sx={{
                      fontSize: { xs: "0.9rem", md: "1rem" },
                    }}
                  />
                }
                text={props.nationality}
              />
            </Grid>
          </Grid>
        </Grid>
      </CardActionArea>
    </StyledPlayerCard>
  );
};
