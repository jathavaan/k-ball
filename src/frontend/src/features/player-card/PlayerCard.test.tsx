import { render, screen } from "@testing-library/react";
import { PlayerCardProps } from "./playerCard.types.ts";
import { PlayerCard } from "./PlayerCard";

const mockPlayerCardProps: PlayerCardProps = {
  playerId: 2890,
  name: "Jo Hyeon-Woo",
  team: "Ulsan Hyundai FC",
  imageUrl: "https://media.api-sports.io/football/players/2890.png",
  position: "Goalkeeper",
  nationality: "Korea Republic",
  age: 33,
};

describe("PlayerCard", () => {
  it("Should render the player name", () => {
    render(<PlayerCard {...mockPlayerCardProps} />);
    expect(screen.getByText(mockPlayerCardProps.name)).toBeDefined();
  });

  it("Should render the player team", () => {
    render(<PlayerCard {...mockPlayerCardProps} />);
    expect(screen.getByText(mockPlayerCardProps.team)).toBeDefined();
  });

  it("Should render the player position", () => {
    render(<PlayerCard {...mockPlayerCardProps} />);
    expect(screen.getByText(mockPlayerCardProps.position)).toBeDefined();
  });

  it("Should render the player nationality", () => {
    render(<PlayerCard {...mockPlayerCardProps} />);
    expect(screen.getByText(mockPlayerCardProps.nationality)).toBeDefined();
  });

  it("Should render player age", () => {
    render(<PlayerCard {...mockPlayerCardProps} />);
    expect(screen.getByText(mockPlayerCardProps.age)).toBeDefined();
  });

  it("Should match PlayerCard snapshot", () => {
    const { asFragment } = render(<PlayerCard {...mockPlayerCardProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
