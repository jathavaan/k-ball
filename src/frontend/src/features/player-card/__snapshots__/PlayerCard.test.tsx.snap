// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerCard > Should match PlayerCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-e6mcx0-MuiPaper-root-MuiCard-root"
    data-testid="player-card"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <a
      class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
      href="/2890"
      tabindex="0"
    >
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-1qw9nqc-MuiGrid2-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-5 css-1tcukf-MuiGrid2-root"
        >
          <img
            alt="Portrait of Jo Hyeon-Woo"
            class="css-1sws5au"
            src="https://media.api-sports.io/football/players/2890.png"
          />
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-hvt765-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-1gp0puz-MuiTypography-root"
            >
              Jo Hyeon-Woo
            </h4>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
              text="Ulsan Hyundai FC"
              title="Team"
            >
              <b>
                Team:
              </b>
               Ulsan Hyundai FC
            </p>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
              text="Goalkeeper"
              title="Pos."
            >
              <b>
                Pos.:
              </b>
               Goalkeeper
            </p>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
              text="Korea Republic"
              title="Nat."
            >
              <b>
                Nat.:
              </b>
               Korea Republic
            </p>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
              text="33"
              title="Age"
            >
              <b>
                Age:
              </b>
               33
            </p>
          </div>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
    </a>
  </div>
</DocumentFragment>
`;

exports[`PlayerCard > Should match PlayerCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-e6mcx0-MuiPaper-root-MuiCard-root"
    data-testid="player-card"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-1qw9nqc-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-5 css-1tcukf-MuiGrid2-root"
      >
        <img
          alt="Portrait of Jo Hyeon-Woo"
          class="css-1sws5au"
          src="https://media.api-sports.io/football/players/2890.png"
        />
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-hvt765-MuiGrid2-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1gp0puz-MuiTypography-root"
          >
            Jo Hyeon-Woo
          </h4>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
            text="Ulsan Hyundai FC"
            title="Team"
          >
            <b>
              Team:
            </b>
             Ulsan Hyundai FC
          </p>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
            text="Goalkeeper"
            title="Pos."
          >
            <b>
              Pos.:
            </b>
             Goalkeeper
          </p>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
            text="Korea Republic"
            title="Nat."
          >
            <b>
              Nat.:
            </b>
             Korea Republic
          </p>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1isr79v-MuiGrid2-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-x09oxf-MuiTypography-root"
            text="33"
            title="Age"
          >
            <b>
              Age:
            </b>
             33
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
